libs/core/algorithms/include/hpx/parallel/algorithms/detail/reduce.hpp

```diff
@@ -0,0 +1,113 @@
+
+    template <typename ExPolicy>
+    struct sequential_reduce_t final
+      : hpx::functional::detail::tag_fallback<sequential_reduce_t<ExPolicy>>
+    {
+    private:
+        template <typename InIterB, typename InIterE, typename T,
+            typename Reduce>
+        friend inline constexpr T tag_fallback_invoke(sequential_reduce_t,
+            ExPolicy&& policy, InIterB first, InIterE last, T init, Reduce&& r)
+        {
+            util::loop_ind(HPX_FORWARD(ExPolicy, policy), first, last,
+                [&init, reduce_op = HPX_FORWARD(Reduce, r)](
+                    const auto& v) mutable {
+                    init = HPX_INVOKE(reduce_op, init, v);
+                });
+            return init;
+        }
+
+        template <typename T, typename FwdIterB, typename Reduce>
+        friend inline constexpr T tag_fallback_invoke(sequential_reduce_t,
+            FwdIterB part_begin, std::size_t part_size, T init, Reduce r)
+        {
+            util::loop_n_ind<ExPolicy>(
+                part_begin, part_size, [&r, &init](const auto& v) mutable {
+                    init = HPX_INVOKE(r, init, v);
+                });
+            return init;
+        }
+
+        template <typename Iter, typename Sent, typename T, typename Reduce,
+            typename Convert>
+        friend inline constexpr auto tag_fallback_invoke(sequential_reduce_t,
+            ExPolicy&& policy, Iter first, Sent last, T init, Reduce&& r,
+            Convert&& conv)
+        {
+            util::loop_ind(HPX_FORWARD(ExPolicy, policy), first, last,
+                [&init, reduce_op = HPX_FORWARD(Reduce, r),
+                    conv_op = HPX_FORWARD(Convert, conv)](
+                    const auto& v) mutable {
+                    auto cnv = conv_op(v);
+                    init = HPX_INVOKE(reduce_op, init, cnv);
+                });
+            return init;
+        }
+
+        template <typename T, typename Iter, typename Reduce, typename Convert>
+        friend inline constexpr auto tag_fallback_invoke(sequential_reduce_t,
+            Iter part_begin, std::size_t part_size, T init, Reduce r,
+            Convert conv)
+        {
+            util::loop_n_ind<ExPolicy>(part_begin, part_size,
+                [&r, &conv, &init](const auto& v) mutable {
+                    auto cnv = conv(v);
+                    init = HPX_INVOKE(r, init, cnv);
+                });
+            return init;
+        }
+
+        template <typename Iter1, typename Sent, typename Iter2, typename T,
+            typename Reduce, typename Convert>
+        friend inline constexpr T tag_fallback_invoke(sequential_reduce_t,
+            Iter1 first1, Sent last1, Iter2 first2, T init, Reduce&& r,
+            Convert&& conv)
+        {
+            util::loop2<ExPolicy>(first1, last1, first2,
+                [&init, reduce_op = HPX_FORWARD(Reduce, r),
+                    convert_op = HPX_FORWARD(Convert, conv)](
```

Comment:
A similar construct above captures the function objects by reference, here you embedded them into the lambda. Both solutions are probably ok (if we're sure the arguments don't go out of scope), I'd suggest however to stay consistent.